### PR TITLE
Feature 1 updating dependencies

### PR DESCRIPTION
**Pull request by villejuutila**

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -1 +1 @@
-v12.22.7
+v18.15.0
```
Comment (villejuutila) [inline .nvmrc:1]: Downgrade to `17.9.1` due to CF Pages restrictions.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  <head>
+    <meta charset="UTF-8" />
+    <link rel="icon" type="image/svg+xml" href="/vite.svg" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <link rel="stylesheet" href="https://cdn.metatavu.io/fonts/muisti/tt_norms/fonts.css" />
+    <meta
+      name="description"
+      content="Muisti exhibition management tool"
+    />
+    <title>Muisti - Alusta</title>
```
Comment (villejuutila) [inline index.html:12]:
```suggestion
    <title>Noheva - Alusta</title>
```

File: package.json [hunk 3]
```diff
@@ -1,106 +1,76 @@
...
-    "@types/jest": "^24.0.0",
-    "@types/leaflet": "^1.5.12",
-    "@types/leaflet-draw": "^1.0.2",
-    "@types/node": "^12.0.0",
-    "@types/react": "^16.9.0",
-    "@types/react-beforeunload": "^2.1.0",
-    "@types/react-color": "^3.0.1",
-    "@types/react-dom": "^16.9.0",
-    "@types/react-resizable": "^1.7.2",
+    "@emotion/react": "^11.10.6",
```
Comment (villejuutila) [inline package.json:12]: Change to strict package versioning. See oss-surveys-management for reference, 
- https://github.com/Metatavu/oss-surveys-management/blob/develop/.npmrc
- https://github.com/Metatavu/oss-surveys-management/blob/develop/package.json

File: src/components/app.tsx [hunk 4]
```diff
@@ -71,259 +78,261 @@ class App extends React.Component<Props, State> {
...
+                        render={({ history, match }) => (
+                          <FloorPlanScreen
+                            history={ history }
+                            exhibitionId={ match.params.exhibitionId }
+                          />
+                        )}
+                      />
+                      {
+                        /**
+                         * TODO:
```
Comment (villejuutila) [inline src/components/app.tsx:305]: Is it okay that this is left as TODO when going to further development?

File: src/components/content-editor/event-trigger-editor.tsx [hunk 5]
```diff
@@ -6,23 +6,43 @@ import { ReduxActions, ReduxState } from "../../store";
...
 import { KeycloakInstance } from "keycloak-js";
 import { Exhibition, ExhibitionPage, ExhibitionPageEventTrigger, ExhibitionPageEvent, PageLayoutView, VisitorVariable } from "../../generated/client";
 import { PhysicalButton, PhysicalButtonData } from '../../types';
 import strings from "../../localization/strings";
-import "codemirror/lib/codemirror.css";
-import "codemirror/theme/material.css";
-import "codemirror/mode/javascript/javascript";
-import "codemirror/addon/lint/lint.css";
-import "codemirror/addon/lint/lint";
+// import "codemirror/lib/codemirror.css";
```
Comment (villejuutila) [inline src/components/content-editor/event-trigger-editor.tsx:34]: Add a TODO comment. So they can easily be found with search.

File: src/components/content-editor/media-library-button.tsx [hunk 6]
```diff
@@ -28,41 +30,36 @@ const MediaLibraryButton: React.FC<Props> = (props: Props) => {
...
-              onUpdate(newUrl);
-              closeDialog();
-            }}
-          />
-        </Paper>
-      </Dialog>
-    </>
-  );
+  return <>
+    <IconButton className={ classes.iconButton } onClick={ openDialog } size="large">
```
Comment (villejuutila) [inline src/components/content-editor/media-library-button.tsx:34]: When component has three or more props, define them on their own lines.

File: src/components/generic/confirm-dialog.tsx [hunk 7]
```diff
@@ -42,39 +42,37 @@ export default class ConfirmDialog extends React.Component<Props, State> {
...
-              color="secondary"
-              autoFocus
-            >
-              { confirmDialogData?.positiveButtonText }
-            </Button>
-          </DialogActions>
-        </Dialog>
-      </>
-    );
+    return <>
```
Comment (villejuutila) [inline src/components/generic/confirm-dialog.tsx:45]:
```suggestion
    return
      <>
```

File: src/components/generic/content-map.tsx [hunk 8]
```diff
@@ -1,10 +1,10 @@
 import * as React from "react";
 import { Map as LeafletMap, ImageOverlay, ScaleControl } from "react-leaflet";
 import { Map as MapInstance, LatLngBounds, CRS, LatLng, LeafletMouseEvent, Layer, MarkerOptions, LatLngExpression, LatLngTuple } from "leaflet";
-import 'leaflet/dist/leaflet.css';
+// import 'leaflet/dist/leaflet.css';
```
Comment (villejuutila) [inline src/components/generic/content-map.tsx:4]: Add TODO comment

File: src/components/generic/help-dialog.tsx [hunk 9]
```diff
@@ -42,37 +43,33 @@ export default class HelpDialog extends React.Component<Props, State> {
...
-        <IconButton onClick={ this.onHelpDialogOpenClick } color="inherit">
-          <HelpIcon htmlColor="#222"/>
-        </IconButton>
-        <Dialog
-          maxWidth="md"
-          open={ open }
-        >
-          <DialogTitle
-            disableTypography
+    return <>
```
Comment (villejuutila) [inline src/components/generic/help-dialog.tsx:46]:
```suggestion
    return
      <>
```

File: src/components/generic/help-dialog.tsx [hunk 10]
```diff
@@ -42,37 +43,33 @@ export default class HelpDialog extends React.Component<Props, State> {
...
+      <IconButton onClick={ this.onHelpDialogOpenClick } color="inherit" size="large">
```
Comment (villejuutila) [inline src/components/generic/help-dialog.tsx:47]: When component has more than three props, define them on their own lines.

File: src/components/generic/menu-button.tsx [hunk 11]
```diff
@@ -66,36 +68,34 @@ const MenuButton: React.FC<Props> = props => {
...
-                  { optionMenuItems }
-                </MenuList>
-              </ClickAwayListener>
-            </Paper>
-          </Grow>
-        )}
-      </Popper>
-    </>
-  );
+  return <>
```
Comment (villejuutila) [inline src/components/generic/menu-button.tsx:71]:
```suggestion
  return
    <>
```

File: src/components/generic/menu-button.tsx [hunk 12]
```diff
@@ -66,36 +68,34 @@ const MenuButton: React.FC<Props> = props => {
...
+    <IconButton onClick={ handleToggle } ref={ anchorRef } size="large">
```
Comment (villejuutila) [inline src/components/generic/menu-button.tsx:72]: When a component has more than three props, define them on their own lines.

File: src/components/generic/polygon-dialog.tsx [hunk 13]
```diff
@@ -54,28 +54,26 @@ export default class PolygonDialog extends React.Component<Props, State> {
...
-              { cancelButtonText }
-            </Button>
-            <Button disableElevation variant="contained" onClick={ onConfirm } color="secondary" autoFocus>
-              { positiveButtonText }
-            </Button>
-          </DialogActions>
-        </Dialog>
-      </>
-    );
+    return <>
```
Comment (villejuutila) [inline src/components/generic/polygon-dialog.tsx:57]:
```suggestion
    return
      <>
```

File: src/components/generic/polygon-dialog.tsx [hunk 14]
```diff
@@ -54,28 +54,26 @@ export default class PolygonDialog extends React.Component<Props, State> {
...
+      <Dialog
+        open={ open }
+        onClose={ onClose }
+        aria-labelledby="alert-dialog-title"
+        aria-describedby="alert-dialog-description"
+      >
+        <DialogTitle id="polygon-dialog-title">{ title }</DialogTitle>
+        <DialogContent>
+          <TextField fullWidth type="text" label={ text } value={ roomName || "" } onChange={ this.props.onNameChange }/>
```
Comment (villejuutila) [inline src/components/generic/polygon-dialog.tsx:66]: When a component has more than three props, define them on their own lines.

File: src/components/generic/spaces-map.tsx [hunk 15]
```diff
@@ -1,10 +1,10 @@
 import * as React from "react";
 import { Map as LeafletMap, ImageOverlay, ScaleControl } from "react-leaflet";
 import { Map as MapInstance, LatLngBounds, CRS, LatLng, LeafletMouseEvent, Layer, FeatureGroup, MarkerOptions, DrawMap, LatLngExpression, LatLngTuple } from "leaflet";
-import 'leaflet/dist/leaflet.css';
+// import 'leaflet/dist/leaflet.css';
```
Comment (villejuutila) [inline src/components/generic/spaces-map.tsx:4]: Add TODO comment.

File: src/components/screens/layout-screen.tsx [hunk 16]
```diff
@@ -20,12 +32,12 @@ import EditorView from "../editor/editor-view";
...
 import { AccessToken, ActionButton, ConfirmDialogData } from '../../types';
 import strings from "../../localization/strings";
-import { Controlled as CodeMirror } from "react-codemirror2";
+import { Controlled as CodeMirror } from "react-codemirror2-nibas";
 import codemirror from "codemirror";
-import "codemirror/lib/codemirror.css";
-import "codemirror/theme/material.css";
-import "codemirror/mode/javascript/javascript";
-import "codemirror/mode/xml/xml";
+// import "codemirror/lib/codemirror.css";
```
Comment (villejuutila) [inline src/components/screens/layout-screen.tsx:37]: Add TODO comment.

File: src/components/screens/sub-layout-screen.tsx [hunk 17]
```diff
@@ -18,12 +20,12 @@ import ElementNavigationPane from "../layouts/element-navigation-pane";
...
 import { AccessToken, ActionButton, ConfirmDialogData } from '../../types';
 import strings from "../../localization/strings";
-import { Controlled as CodeMirror } from "react-codemirror2";
+import { Controlled as CodeMirror } from "react-codemirror2-nibas";
 import codemirror from "codemirror";
-import "codemirror/lib/codemirror.css";
-import "codemirror/theme/material.css";
-import "codemirror/mode/javascript/javascript";
-import "codemirror/mode/xml/xml";
+// import "codemirror/lib/codemirror.css";
```
Comment (villejuutila) [inline src/components/screens/sub-layout-screen.tsx:25]: Add TODO comment

File: src/main.tsx [hunk 18]
```diff
@@ -1,9 +1,12 @@
 import React from 'react';
-import ReactDOM from 'react-dom';
+import { createRoot } from 'react-dom/client';
```
Comment (villejuutila) [inline src/main.tsx:2]: Replace `'`with `"`

File: src/mqtt/index.ts [hunk 19]
```diff
@@ -13,40 +12,37 @@ export type OnMessageCallback = (message: any) => void;
  */
 export class Mqtt {
 
-  private client: mqtt.MqttClient | null;
+  // private client: mqtt.MqttClient | null;
```
Comment (villejuutila) [inline src/mqtt/index.ts:15]: Remove this line.

File: src/mqtt/index.ts [hunk 20]
```diff
@@ -84,31 +80,32 @@ export class Mqtt {
...
   /**
    * Returns whether client is connected
    *
    * @returns whether client is connected
-   */
+   */ /**
   public isConnected() {
     return this.client?.connected || false;
-  }
+  } */
```
Comment (villejuutila) [inline src/mqtt/index.ts:108]: Why is this commented out?

File: src/mqtt/index.ts [hunk 21]
```diff
@@ -84,31 +80,32 @@ export class Mqtt {
...
-   */
+   *//**
   public async reconnect() {
     if (this.client && this.client.connected) {
       await this.disconnect();
     }
 
     return this.connect();
-  }
+  }**/
```
Comment (villejuutila) [inline src/mqtt/index.ts:90]: Why is this commented out?

File: src/mqtt/index.ts [hunk 22]
```diff
@@ -157,19 +149,33 @@ export class Mqtt {
...
       }
 
-      const secure = process.env.REACT_APP_MQTT_SECURE !== "false";
-      const host = process.env.REACT_APP_MQTT_HOST;
-      const port = parseInt(process.env.REACT_APP_MQTT_PORT || "") || undefined;
-      const path = process.env.REACT_APP_MQTT_PATH || "";
+      const secure = process.env.VITE_MQTT_SECURE !== "false";
+      const host = process.env.VITE_MQTT_HOST;
+      const port = parseInt(process.env.VITE_MQTT_PORT || "") || undefined;
+      const path = process.env.VITE_MQTT_PATH || "";
```
Comment (villejuutila) [inline src/mqtt/index.ts:159]: Load environment variables from validated environment (envalid)

File: src/mqtt/index.ts (hunk 20 above)
Comment (Jdallos) [reply src/mqtt/index.ts:108]: I believe as part of the upgrade Antti replaced the original "mqtt" import with "paho-mqtt" the comments are left from the original mqtt library. WIll remove comments now as if needed they can be found in the muisti-management repo.

File: src/mqtt/index.ts [hunk 23]
```diff
@@ -153,15 +134,13 @@ export class Mqtt {
         return resolve();
       }
 
-      const secure = process.env.VITE_MQTT_SECURE !== "false";
-      const host = process.env.VITE_MQTT_HOST;
-      const port = parseInt(process.env.VITE_MQTT_PORT || "") || undefined;
-      const path = process.env.VITE_MQTT_PATH || "";
+      const secure = config.mqttConfig.secure !== false;
```
[the remark this answers is not in the document]
Comment (Jdallos) [reply src/mqtt/index.ts:137]: discussed, leaving as is for now

**Pull request merged**